app/src/main/java/ch/epfl/sweng/eventmanager/notifications/ScheduledItemNotification.java

```diff
@@ -0,0 +1,108 @@
+        Intent intent = new Intent(context, NotificationPublisher.class);
+        PendingIntent pendingIntent = PendingIntent.getBroadcast(context.getApplicationContext(), scheduledItem.getId().hashCode(), intent, PendingIntent.FLAG_UPDATE_CURRENT);
+        AlarmManager am = (AlarmManager) context.getApplicationContext().getSystemService(Context.ALARM_SERVICE);
+        am.cancel(pendingIntent);
+        // Cancel the `PendingIntent` after you've canceled the alarm
+        pendingIntent.cancel();
+    }
+
+    /**
+     * Compute the time remaining until {@param scheduledItem} starts
+     *
+     * @param scheduledItem, not null
+     * @return time in milliseconds
+     * @throws NullPointerException  if the date of {@param scheduledItem} is null
+     * @throws IllegalStateException if {@param scheduledItem} is already past
+     */
+    private static long getTimeTo(@NonNull ScheduledItem scheduledItem) {
+        Date scheduledItemDate = scheduledItem.getDate();
+        if (scheduledItemDate == null)
+            throw new NullPointerException();
+        long timeUntil = scheduledItem.getDate().getTime() - System.currentTimeMillis();
+
+        if (timeUntil < 0) // Event is past
+            throw new IllegalStateException();
+        else return timeUntil;
+    }
+
+    private static Notification getNotification(@NonNull Context context, ScheduledItem scheduledItem) {
+        // Set up on tap action TODO display my_schedule fragment when tap on notification
+        Intent intent = new Intent(context, EventPickingActivity.class);
+        intent.setFlags(Intent.FLAG_ACTIVITY_NEW_TASK | Intent.FLAG_ACTIVITY_CLEAR_TASK);
+        PendingIntent pendingIntent = PendingIntent.getActivity(context, 0, intent, 0);
+
+        // Create a notification
+        NotificationCompat.Builder builder = new NotificationCompat.Builder(context, CHANNEL_ID)
+                .setContentTitle(scheduledItem.getArtist() + " concert will start soon")
+                .setContentText(scheduledItem.getDescription())
+                .setSmallIcon(R.drawable.ic_marker_active) // TODO define a small icon for notifications
+                .setPriority(NotificationCompat.PRIORITY_DEFAULT)
+                .setAutoCancel(true)
+                .setContentIntent(pendingIntent);
+
+        return builder.build();
+    }
+
+    private static void setupNotificationChannel(@NonNull Context context) {
+        NotificationManager notificationManager = (NotificationManager) context.getSystemService(Context.NOTIFICATION_SERVICE);
+
+        // Setup Notification Channel
+        if (Build.VERSION.SDK_INT >= Build.VERSION_CODES.O) {
+            NotificationChannel channel = new NotificationChannel(CHANNEL_ID,
+                    CHANNEL_NAME,
+                    NotificationManager.IMPORTANCE_DEFAULT);
+            notificationManager.createNotificationChannel(channel);
+        }
```

Comment:
Does it work for lower versions than O ? 
If not do you think a TODO handle other versions is relevant ?

Reply:
This is specific to SDK > 28 and not relevant for other versions.